pkg/cluster/fixssh.go

```diff
@@ -124,52 +161,117 @@ func (m *manager) removeBackendPoolsFromNIC(ctx context.Context, resourceGroup,
+func (m *manager) updateILBAddressPool(ctx context.Context, nic *mgmtnetwork.Interface, nicName string, lb *mgmtnetwork.LoadBalancer, i int, infraID string) bool {
 	if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
 		m.log.Warnf("unable to update NIC as there are no IP configurations for %s", nicName)
 		return false
 	}
 
 	var ilbBackendPool string
+
 	switch m.doc.OpenShiftCluster.Properties.ArchitectureVersion {
 	case api.ArchitectureVersionV1:
 		ilbBackendPool = infraID + "-internal-controlplane-v4"
 	case api.ArchitectureVersionV2:
 		ilbBackendPool = infraID
 	}
 
-	sshID := fmt.Sprintf("%s/backendAddressPools/ssh-%d", *lb.ID, i)
-	ilbID := fmt.Sprintf("%s/backendAddressPools/%s", *lb.ID, ilbBackendPool)
+	sshBackendPoolID := fmt.Sprintf("%s/backendAddressPools/ssh-%d", *lb.ID, i)
+	ilbBackendPoolID := fmt.Sprintf("%s/backendAddressPools/%s", *lb.ID, ilbBackendPool)
 
 	updateSSHPool := true
 	updateILBPool := true
 
-	for _, p := range *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools {
-		if strings.EqualFold(*p.ID, sshID) {
-			updateSSHPool = false
-		}
-		if strings.EqualFold(*p.ID, ilbID) {
-			updateILBPool = false
+	ipc := (*nic.InterfacePropertiesFormat.IPConfigurations)[0]
+	if ipc.LoadBalancerBackendAddressPools == nil {
+		emptyBackendAddressPool := make([]mgmtnetwork.BackendAddressPool, 0)
+		(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = &emptyBackendAddressPool
+	} else {
+		for _, p := range *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools {
+			if strings.EqualFold(*p.ID, sshBackendPoolID) {
+				updateSSHPool = false
+			}
+			if strings.EqualFold(*p.ID, ilbBackendPoolID) {
+				updateILBPool = false
+			}
 		}
 	}
 
 	if updateSSHPool {
-		m.log.Printf("Adding NIC %s to Internal Load Balancer SSH Backend Address Pool %s", nicName, sshID)
+		m.log.Printf("Adding NIC %s to Internal Load Balancer SSH Backend Address Pool %s", nicName, sshBackendPoolID)
 		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
-			ID: &sshID,
+			ID: &sshBackendPoolID,
 		})
 	}
 
 	if updateILBPool {
-		m.log.Printf("Adding NIC %s to Internal Load Balancer API Address Pool %s", nicName, ilbID)
+		m.log.Printf("Adding NIC %s to Internal Load Balancer API Address Pool %s", nicName, ilbBackendPoolID)
 		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
-			ID: &ilbID,
+			ID: &ilbBackendPoolID,
 		})
 	}
 
 	return updateSSHPool || updateILBPool
 }
 
+func (m *manager) updateV1ELBAddressPool(ctx context.Context, nic *mgmtnetwork.Interface, nicName string, lb *mgmtnetwork.LoadBalancer, i int, infraID string) bool {
```

Comment:
`i` is not used here, `lb` does not need to be passed in, if we change `updateV1ELBAddressPool` to accept a `resourceGroup string` and to return a `error` instead of `bool` the first lines of the method can retrieve the `lb` and we can clean up the switch to just call `updateV1ELBAddressPool` and return its possible error.